### PR TITLE
[fix/ocis-disable-resharing] Remove (re)sharing from roles for newly created user/group shares for ocis

### PR DESCRIPTION
**Pull request by felix-schwarz**
## Description
Removes (re)sharing from roles for newly created user/group shares for ocis.

## Related Issue
https://github.com/owncloud/ios-app/issues/1375

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: ownCloudSDK/Core/Sharing/OCCore+Sharing.m [hunk 1]
```diff
@@ -383,12 +384,12 @@ - (nullable NSProgress *)makeDecisionOnShare:(OCShare *)share accept:(BOOL)accep
...
 				[[OCShareRole alloc] initWithType:OCShareRoleTypeViewer
 						       shareTypes:OCShareTypesMaskUserShare|OCShareTypesMaskGroupShare
-						     permissions:OCSharePermissionsMaskRead|OCSharePermissionsMaskShare
+						     permissions:OCSharePermissionsMaskRead|(resharingSupported ? OCSharePermissionsMaskShare : 0)
 					 customizablePermissions:OCSharePermissionsMaskNone
 						       locations:OCLocationTypeFile|OCLocationTypeFolder
 						      symbolName:@"eye.fill"
 						   localizedName:OCLocalized(@"Viewer")
-					    localizedDescription:OCLocalized(@"Download, preview and share")],
+					    localizedDescription:OCLocalized(resharingSupported ? @"Download, preview and share" : @"Download and preview")],
```
Comment (hosy) [inline ownCloudSDK/Core/Sharing/OCCore+Sharing.m:393]: `OCLocalized` needs to be changed to `OCLocalizedString` to get in sync with the new translation changes before merging.

File: ownCloudSDK/Core/Sharing/OCCore+Sharing.m [hunk 2]
```diff
@@ -404,22 +405,22 @@ - (nullable NSProgress *)makeDecisionOnShare:(OCShare *)share accept:(BOOL)accep
...
 				[[OCShareRole alloc] initWithType:OCShareRoleTypeEditor
 						       shareTypes:OCShareTypesMaskUserShare|OCShareTypesMaskGroupShare
-						     permissions:OCSharePermissionsMaskRead|OCSharePermissionsMaskUpdate|OCSharePermissionsMaskShare
+						     permissions:OCSharePermissionsMaskRead|OCSharePermissionsMaskUpdate|(resharingSupported ? OCSharePermissionsMaskShare : 0)
 					 customizablePermissions:OCSharePermissionsMaskNone
 						       locations:OCLocationTypeFile
 						      symbolName:@"pencil"
 						   localizedName:OCLocalized(@"Editor")
-					    localizedDescription:OCLocalized(@"Edit, download, preview and share")],
+					    localizedDescription:OCLocalized(resharingSupported ? @"Edit, download, preview and share" : @"Edit, download and preview")],
```
Comment (hosy) [inline ownCloudSDK/Core/Sharing/OCCore+Sharing.m:414]: `OCLocalized` needs to be changed to `OCLocalizedString` to get in sync with the new translation changes before merging.

File: ownCloudSDK/Core/Sharing/OCCore+Sharing.m [hunk 3]
```diff
@@ -404,22 +405,22 @@ - (nullable NSProgress *)makeDecisionOnShare:(OCShare *)share accept:(BOOL)accep
...
 				[[OCShareRole alloc] initWithType:OCShareRoleTypeEditor
 						       shareTypes:OCShareTypesMaskUserShare|OCShareTypesMaskGroupShare
-						     permissions:OCSharePermissionsMaskRead|OCSharePermissionsMaskUpdate|OCSharePermissionsMaskCreate|OCSharePermissionsMaskDelete|OCSharePermissionsMaskShare
+						     permissions:OCSharePermissionsMaskRead|OCSharePermissionsMaskUpdate|OCSharePermissionsMaskCreate|OCSharePermissionsMaskDelete|(resharingSupported ? OCSharePermissionsMaskShare : 0)
 					 customizablePermissions:OCSharePermissionsMaskNone
 						       locations:OCLocationTypeFolder
 						      symbolName:@"pencil"
 						   localizedName:OCLocalized(@"Editor")
-					    localizedDescription:OCLocalized(@"Upload, edit, delete, download, preview and share")],
+					    localizedDescription:OCLocalized(resharingSupported ? @"Upload, edit, delete, download, preview and share" : @"Upload, edit, delete, download and preview")],
```
Comment (hosy) [inline ownCloudSDK/Core/Sharing/OCCore+Sharing.m:424]: `OCLocalized` needs to be changed to `OCLocalizedString` to get in sync with the new translation changes before merging.

File: ownCloudSDK/Core/Sharing/OCCore+Sharing.m [hunk 4]
```diff
@@ -435,19 +436,19 @@ - (nullable NSProgress *)makeDecisionOnShare:(OCShare *)share accept:(BOOL)accep
...
 				[[OCShareRole alloc] initWithType:OCShareRoleTypeManager
 						       shareTypes:OCShareTypesMaskUserShare|OCShareTypesMaskGroupShare
-						     permissions:OCSharePermissionsMaskRead|OCSharePermissionsMaskUpdate|OCSharePermissionsMaskCreate|OCSharePermissionsMaskDelete|OCSharePermissionsMaskShare
+						     permissions:OCSharePermissionsMaskRead|OCSharePermissionsMaskUpdate|OCSharePermissionsMaskCreate|OCSharePermissionsMaskDelete|(resharingSupported ? OCSharePermissionsMaskShare : 0)
 					 customizablePermissions:OCSharePermissionsMaskNone
 						       locations:OCLocationTypeDrive
 						      symbolName:@"person.fill"
 						   localizedName:OCLocalized(@"Manager")
-					    localizedDescription:OCLocalized(@"Upload, edit, delete, download, preview and share")],
+					    localizedDescription:OCLocalized(resharingSupported ? @"Upload, edit, delete, download, preview and share" : @"Upload, edit, delete, download and preview")],
```
Comment (hosy) [inline ownCloudSDK/Core/Sharing/OCCore+Sharing.m:445]: `OCLocalized` needs to be changed to `OCLocalizedString` to get in sync with the new translation changes before merging.

**Review by hosy — CHANGES_REQUESTED** [review]

File: ownCloudSDK/Core/Sharing/OCCore+Sharing.m (hunk 1 above)
Comment (felix-schwarz) [reply ownCloudSDK/Core/Sharing/OCCore+Sharing.m:393]: Fixed.

File: ownCloudSDK/Core/Sharing/OCCore+Sharing.m (hunk 2 above)
Comment (felix-schwarz) [reply ownCloudSDK/Core/Sharing/OCCore+Sharing.m:414]: Fixed.

File: ownCloudSDK/Core/Sharing/OCCore+Sharing.m (hunk 3 above)
Comment (felix-schwarz) [reply ownCloudSDK/Core/Sharing/OCCore+Sharing.m:424]: Fixed.

File: ownCloudSDK/Core/Sharing/OCCore+Sharing.m (hunk 4 above)
Comment (felix-schwarz) [reply ownCloudSDK/Core/Sharing/OCCore+Sharing.m:445]: Fixed.

**Review by hosy — APPROVED** [review]

**Pull request merged**